runtime/error.c

```diff
@@ -116,13 +128,26 @@ static int storePC(void *data, PC pc) {
     return SUCCESS;
 }
 
-void _bal_error_backtrace_print(ErrorPtr ep) {
+void _bal_error_backtrace_print(ErrorPtr ep, uint32_t start) {
     GC PC *pcs = ep->pcs;
     uint32_t nPCs = ep->nPCs;
     BacktraceError err = { ep->backtraceErrorCode, ep->backtraceErrorMessage };
-    for (uint32_t i = 0; i < nPCs; i++) {
-        backtrace_pcinfo(state, pcs[i], printBacktraceLine, storeBacktraceError, &err);
+    if (start < nPCs) {
+        BacktraceStartLine backtraceStartLine = { err, ep->lineNumber };
```

Comment:
I would separate everything in this if branch into a separate function.

Reply:
Changed